edb/ir/utils.py

```diff
@@ -451,3 +451,21 @@ def contains_set_of_op(ir: irast.Base) -> bool:
     flt = (lambda n: any(x == ft.TypeModifier.SetOfType
                          for x in n.params_typemods))
     return bool(ast.find_children(ir, irast.Call, flt, terminate_early=True))
+
+
+class PointerCollectorVisitor(ast.NodeVisitor):
+    collected: List[irast.Pointer]
+
+    def __init__(self) -> None:
+        super().__init__()
+        self.collected = []
+
+    def visit_Pointer(self, pointer: irast.Pointer) -> None:
+        self.collected.append(pointer)
+
+
+def collect_pointers(expr: irast.Base) -> List[irast.Pointer]:
```

Comment:
I think you can use `find_children` instead of defining a visitor.
(And at that point, it's probably best just to inline the call to `find_children`)